fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/LoanDelinquencyDomainServiceImpl.java

```diff
@@ -48,12 +48,15 @@ public CollectionData getOverdueCollectionData(final Loan loan) {
         boolean oldestOverdueInstallment = false;
         boolean overdueSinceDateWasSet = false;
         boolean firstNotYetDueInstallment = false;
+        LoanRepaymentScheduleInstallment latestInstallment = loan.getRepaymentScheduleInstallments()
+                .get(loan.getRepaymentScheduleInstallments().size() - 1);
+
+        List<LoanTransaction> chargebackTransactions = loan.getLoanTransactions().stream().filter(LoanTransaction::isChargeback).toList();
```

Comment:
Same thing as above, usability perspective, let's move this into the Loan to get only the chargeBack transactions. Even better if you make it more abstract and do like this:

```
loan.getLoanTransactions(LoanTransaction::isChargeback);
```